beamer/agent/util.py

```diff
@@ -66,9 +66,16 @@ def transact(
     except (ContractLogicError, requests.exceptions.RequestException) as exc:
         raise TransactionFailed() from exc
 
-    receipt = func.w3.eth.wait_for_transaction_receipt(
-        txn_hash, timeout=timeout, poll_latency=poll_latency
-    )
+    while True:
+        try:
+            receipt = func.w3.eth.wait_for_transaction_receipt(
+                txn_hash, timeout=timeout, poll_latency=poll_latency
+            )
+        except TimeExhausted as exc:
+            log.error("timed out waiting for tx receipt, retrying", exc=exc)
```

Comment:
nitpick: If it's not a name like function name we normally start logs with capital letters

Reply:
Fixed, thanks!